### PR TITLE
Added ability targets to display in casts table

### PR DESCRIPTION
**Pull request by Argn0**
fixes https://github.com/odota/web/issues/1172

### CODE REVIEW

File: processors/createParsedDataBlob.js [hunk 1]
```diff
@@ -62,6 +62,7 @@ function getParseSchema() {
         killed: {},
         item_uses: {},
         ability_uses: {},
+        ability_targeted: {},
```
Comment (howardchung) [inline processors/createParsedDataBlob.js]: I think this will only store the ability name and the number of casts?

**Comment by howardchung** [comment]
I think the ask was for the targets of the player's spells, not how many times they were targeted, although maybe it would be possible to calculate the inverse from storing the data like this?

**Comment by Argn0** [comment]
eslint changed counts to count for some reason... :D

File: sql/create_tables.cql [hunk 2]
```diff
@@ -87,6 +87,7 @@ CREATE TABLE IF NOT EXISTS player_matches (
   killed text,
   item_uses text,
   ability_uses text,
+  ability_targets text,
```
Comment (howardchung) [inline sql/create_tables.cql]: can you also add the ALTER TABLE statements to the migrations files?

File: processors/processExpand.js [hunk 3]
```diff
@@ -201,6 +201,13 @@ function processExpand(entries, meta) {
         key: translate(e.inflictor),
         type: 'ability_uses',
       });
+      // target of ability
+      expand({
+        time: e.time,
+        unit: e.attackername,
+        key: [translate(e.inflictor), translate(e.targetname)],
```
Comment (howardchung) [inline processors/processExpand.js]: I think translate checks for illusion and prepends illusion_ if it is

Comment (howardchung) [reply processors/processExpand.js]: not sure if there are any more cases we might want to exclude.  Perhaps we should only track targets on heroes to save space?  or maybe we want to track targets on non-heroes so people can be shamed for dooming creeps etc

**Comment by Argn0** [comment]
matchID 3615247598 https://pastebin.com/m7st7j3C


{
  "axe_battle_hunger": {
    "npc_dota_hero_bristleback": 3,
    "npc_dota_hero_juggernaut": 1,
    "npc_dota_hero_phantom_assassin": 3
  },
  "axe_culling_blade": {
    "npc_dota_hero_juggernaut": 1,
    "npc_dota_hero_shadow_shaman": 1,
    "npc_dota_hero_phantom_assassin": 2
  }
}

________________________________________

{
  "legion_commander_press_the_attack": {
    "npc_dota_hero_legion_commander": 22,
    "npc_dota_hero_axe": 2,
    "npc_dota_hero_sven": 1
  },
  "legion_commander_duel": {
    "npc_dota_hero_phantom_assassin": 2,
    "npc_dota_hero_shadow_shaman": 1,
    "npc_dota_hero_juggernaut": 1,
    "npc_dota_hero_bristleback": 1
  }
}

__________________________________________


{
  "phantom_assassin_stifling_dagger": {
    "npc_dota_hero_zuus": 14,
    "npc_dota_hero_sven": 5,
    "npc_dota_hero_axe": 14,
    "npc_dota_hero_legion_commander": 13,
    "npc_dota_hero_clinkz": 15
  },
  "phantom_assassin_phantom_strike": {
    "npc_dota_hero_axe": 5,
    "npc_dota_hero_legion_commander": 5,
    "npc_dota_hero_clinkz": 8,
    "npc_dota_hero_zuus": 6,
    "npc_dota_hero_shadow_shaman": 2,
    "npc_dota_hero_sven": 4
  }
}

File: processors/processExpand.js (hunk 3 above)
Comment (howardchung) [reply processors/processExpand.js]: Add the `      if (e.targethero && !e.targetillusion) {` here

File: processors/populate.js [hunk 4]
```diff
@@ -72,6 +72,20 @@ function populate(e, container) {
           arrEntry = JSON.parse(JSON.stringify(e));
         }
         t.push(arrEntry);
+      } else if (e.type === 'ability_targets') {
+        // e.g. { Telekinesis: { Antimage: 1, Bristleback: 2 }, Fade Bolt: { Lion: 4, Timber: 5 }, ... }
+        const ability = e.key[0];
+        const target = e.key[1];
+        if (target && target.indexOf('hero') !== -1) { // we only want hero targets for now
```
Comment (howardchung) [inline processors/populate.js]: don't do specific checks in populate, do them in the caller

**Pull request merged**